File: Leetcode/Complete/27.RemoveElement.cs
public class RemoveElement
{
	public int RemoveElement(int[] nums, int val) {
        int endPointer = nums.Length;

        // return 0 value array if 1 num and it's equal to val
        if(endPointer == 1 && nums[0] == val) {nums[0] = 0; return 0;};

		for (int i = nums.Length - 1; i >= 0; i--)
		{
			if (nums[i] == val)
			{
				endPointer--;
				nums[i] = nums[endPointer];			
			}
		}
		return endPointer;
    }
}

File: Go/problems/27.RemoveElement.go
package problems

func RemoveElement(nums []int, val int) (int, []int) {
	p1 := 0 // Read Pointer
	p2 := 0 // Write Pointer

	// loop read pointer through whole array
	for p1 < len(nums) {
		// if value at read pointer is not equal to the val continue
		if nums[p1] != val {
			// Move first good element to start of sorted section
			nums[p2] = nums[p1]
			// increment to next slot for non val ints
			p2++
		}

		p1++
	}
	// return p2 because it's index the exact count of elements that were kept
	// which is the next spot in the array for a good/ non val element
	return p2, nums
}
